### PR TITLE
sys_sm: Implemented sys_sm_shutdown()

### PR DESCRIPTION
**Pull request by brian218**
Attempt to implement sys_sm_shutdown() based on @elad335's suggestion in #12836.

### CODE REVIEW

**Comment by brian218** [comment]
It worked properly for me during test.

File: rpcs3/Emu/Cell/lv2/sys_process.cpp [hunk 1]
```diff
@@ -402,6 +402,22 @@ void _sys_process_exit2(ppu_thread& ppu, s32 status, vm::ptr<sys_exit2_param> ar
...
+void sys_process_exit3(ppu_thread& ppu, s32 status)
+{
+	ppu.state += cpu_flag::wait;
+
+	sys_process.warning("_sys_process_exit3(status=%d)", status);
+
+	return _sys_process_exit(ppu, status, 0, 0);
+}
+
+void _sys_process_restart(ppu_thread& ppu, std::vector<std::string>& argv, std::vector<std::string>& envp, std::vector<u8>& data)
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_process.cpp:417]: maybe call it lv2_exitspawn, and use std::span instead of vectors.

Comment (brian218) [reply rpcs3/Emu/Cell/lv2/sys_process.cpp]: Done.

**Review by elad335 — APPROVED** [review]

**Comment by brian218** [comment]
With this PR successfully merged, the following manual process mentioned in the RPCS3 Wiki will be no longer required as the VSH will automatically reboot after rebuilding the XRegistry database.
Wiki

File: rpcs3/Emu/Cell/lv2/sys_sm.cpp [hunk 2]
```diff
@@ -41,9 +43,40 @@ error_code sys_sm_get_ext_event2(vm::ptr<u64> a1, vm::ptr<u64> a2, vm::ptr<u64>
 	return not_an_error(CELL_EAGAIN);
 }
 
-error_code sys_sm_shutdown(u16 op, vm::ptr<void> param, u64 size)
+error_code sys_sm_shutdown(ppu_thread& ppu, u16 op, vm::ptr<void> param, u64 size)
 {
-	sys_sm.todo("sys_sm_shutdown(op=0x%x, param=*0x%x, size=0x%x)", op, param, size);
+	sys_sm.trace("sys_sm_shutdown(op=0x%x, param=*0x%x, size=0x%x)", op, param, size);
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_sm.cpp:48]: Please avoid using trace in syscalls, it is a workaround for massive log spam. In a perfect world we wouldn't need it. Here it is success.

Comment (brian218) [reply rpcs3/Emu/Cell/lv2/sys_sm.cpp:48]: Will change it.
By the way, should I add `ppu.state += cpu_flag::wait` at the beginning of this syscall?

Comment (elad335) [reply rpcs3/Emu/Cell/lv2/sys_sm.cpp:48]: Yes.

Comment (brian218) [reply rpcs3/Emu/Cell/lv2/sys_sm.cpp:48]: > Please avoid using trace in syscalls, it is a workaround for massive log spam. In a perfect world we wouldn't need it. Here it is success.

We have already had at least one .success() log in switch...case, do we still need to make the call log ("sys_sm_shutdown(op=0x%x, param=*0x%x, size=0x%x)") `success`?

Comment (elad335) [reply rpcs3/Emu/Cell/lv2/sys_sm.cpp:48]: Doesn't hurt, and it has unused params so better to them.

**Review by elad335 — APPROVED** [review]

File: rpcs3/Emu/Cell/lv2/sys_sm.cpp (hunk 2 above)
Comment (brian218) [reply rpcs3/Emu/Cell/lv2/sys_sm.cpp]: Done.

**Comment by brian218** [comment]
I'll rebase the commits when I'm in front of my PC.

**Comment by brian218** [comment]
> With this PR successfully merged, the following manual process mentioned in the RPCS3 Wiki will be no longer required as the VSH will automatically reboot after rebuilding the XRegistry database. Wiki

@AniLeo Could you remove the specific instruction from the wiki given that this PR has been merged and released?

**Pull request merged**